version: '2.4'
services:
  app:
    container_name: "application-server"
    image: "jupyter/datascience-notebook:latest"
    restart: always
    volumes:
      - ..:/workspace:cached
      - work:/home/jupyter
    ports:
      - '8888:8888'
    environment:
      "JUPYTER_ENABLE_LAB": "yes"
      "JUPYTER_TOKEN": "docker"
    networks:
      - app-network
    cap_add:
      - SYS_ADMIN
    devices:
      - /dev/fuse
    security_opt:
      - apparmor:unconfined
networks:
  app-network:
    external: true
volumes:
  work:
